spec/api/configuration/retrieve_spec.rb

```diff
@@ -16,13 +17,21 @@
     end
 
     before(:each) do
+      ::Afterpay.configure do |c|
+        c.user_agent = user_agent
+      end
       stub_request(:get, %r{api.us.afterpay.com/v2/configuration})
+        .with(headers: { "User-Agent" => user_agent })
         .to_return(
           status: 200,
           body: raw_response,
           headers: { 'Content-Type' => 'application/json' }
         )
     end
+    it 'makes request with the correct headers with the user_agent' do
+      expect(described_class.call).to have_requested(:get, "https://api.us.afterpay.com/v2/configuration").
```

Comment:
can you use the regexp here instead of the full URL? `%r{api.us.afterpay.com/v2/configuration}`

Reply:
I did it first but it requires a string